model/common/tests/common/grid/mpi_tests/test_parallel_icon.py

```diff
@@ -124,22 +144,30 @@ def test_distributed_local(
 @pytest.mark.parametrize("processor_props", [True], indirect=True)
 @pytest.mark.mpi
 @pytest.mark.parametrize("dim", utils.main_horizontal_dims())
-@pytest.mark.parametrize("marker", [h_grid.Zone.HALO, h_grid.Zone.HALO_LEVEL_2])
+@pytest.mark.parametrize(
+    "experiment",
+    [
+        test_defs.Experiments.MCH_CH_R04B09,
+    ],
+)
+@pytest.mark.parametrize("zone, level", [(h_grid.Zone.HALO, 1), (h_grid.Zone.HALO_LEVEL_2, 2)])
 def test_distributed_halo(
     processor_props: decomp_defs.ProcessProperties,
     dim: gtx.Dimension,
-    marker: h_grid.Zone,
+    zone: h_grid.Zone,
     icon_grid: base_grid.Grid,
+    experiment: test_defs.Experiment,
+    level: int,
 ) -> None:
-    check_comm_size(processor_props)
-    num = int(next(iter(re.findall(r"\d+", marker.value))))
-    domain = h_grid.domain(dim)(marker)
+    parallel_helpers.check_comm_size(processor_props)
+    domain = h_grid.domain(dim)(zone)
     start_index = icon_grid.start_index(domain)
     end_index = icon_grid.end_index(domain)
     rank = processor_props.rank
     print(
-        f"rank {rank}/{processor_props.comm_size} dim = {dim}  {marker} : ({start_index}, {end_index})"
+        f"rank {rank}/{processor_props.comm_size} dim = {dim}  {zone} : ({start_index}, {end_index})"
     )
-
-    assert start_index == HALO_IDX[processor_props.comm_size][dim][rank][num - 1]
-    assert end_index == HALO_IDX[processor_props.comm_size][dim][rank][num]
+    expected = HALO_IDX[processor_props.comm_size][dim][rank][level - 1]
+    assert start_index == expected, f"expected start index {expected}, but was {start_index}"
```

Comment:
Just curious about this change: does the `assert` not print the values (including which one is `start_index` and `expected`) on failure?

Reply:
not as such, only if you have `assert condition, error_msg`. The error_msg is optional.